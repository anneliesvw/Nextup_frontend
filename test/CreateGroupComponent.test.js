import Vuex from 'vuex';
import { shallow, createLocalVue } from 'vue-test-utils';
import CreateGroupComp from '../src/components/groups/CreateGroup.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('CreateGroupComp', () => {
  let store;
  let actions;
  beforeEach(() => {
    actions = {
      addGroup: jest.fn(),
      createroom: jest.fn(),
    };
    store = new Vuex.Store({
      actions,
    });
  });

  it('dispatches addGroup when a button is clicked', () => {
    const wrapper = shallow(CreateGroupComp, {
      store,
      localVue,
    });
    wrapper.find('.create-group-btn').trigger('click');
    expect(actions.addGroup.mock.calls).toHaveLength(1);
  });

  it('groupInfo should have an empty name and description', () => {
    const wrapper = shallow(CreateGroupComp, {
      store,
      localVue,
    });
    expect(wrapper.vm.groupInfo.name).toBe('');
    expect(wrapper.vm.groupInfo.description).toBe('');
  });

  it('friends should be an empty array', () => {
    const wrapper = shallow(CreateGroupComp, {
      store,
      localVue,
    });
    expect(wrapper.vm.friends).toEqual([]);
  });

  it('members should be an empty array', () => {
    const wrapper = shallow(CreateGroupComp, {
      store,
      localVue,
    });
    expect(wrapper.vm.members).toEqual([]);
  });

  it('memberToAdd should be empty', () => {
    const wrapper = shallow(CreateGroupComp, {
      store,
      localVue,
    });
    expect(wrapper.vm.memberToAdd).toBe('');
  });

  // it('calls createroom when group created', () => {
  //   const wrapper = shallow(CreateGroupComp, {
  //     store,
  //     localVue,
  //   });

  //   wrapper.find('.create-group-btn').trigger('click');
  //   expect(actions.createroom.mock.calls).toHaveLength(1);
  // });
});
