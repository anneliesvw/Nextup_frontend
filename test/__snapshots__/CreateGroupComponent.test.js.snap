// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateGroupComp should match snapshot 1`] = `
<el-dialog
  class="create-group-dialog"
  title="Create group"
>
  <div
    class="create-group-form"
  >
    <el-form
      class="create-group-name"
      label-position="top"
    >
      <!---->
       
      <div
        class="create-group-left"
      >
        <el-form-item
          label="Name"
        >
          <el-input
            class="create-group-input"
          />
        </el-form-item>
         
        <el-form-item
          label="Description"
        >
          <el-input
            autosize="[object Object]"
            class="create-group-input"
            type="textarea"
          />
        </el-form-item>
      </div>
    </el-form>
  </div>
   
  <span
    class="dialog-footer"
    slot="footer"
  >
    <el-button>
      Cancel
    </el-button>
     
    <el-button
      class="create-group-btn"
      type="primary"
    >
      Create group
    </el-button>
  </span>
</el-dialog>
`;
